File: typescript/6kyu/FindTheOddInt.ts
/*
Given an array of integers, find the one that appears an odd number of times.

There will always be only one integer that appears an odd number of times.
*/

export const findOdd = (xs: number[]): number => {
  // initialize the return value
  let retVal: number = xs[0];

  // if the received value is only one character, return it
  if (xs.length === 1) return retVal;

  // iterate over the received number
  for (let i: number = 0; i < xs.length; i++) {
    // initialize a count
    let cnt: number = 0;

    // // iterate over the received number
    for (let j: number = 0; j < xs.length; j++) {
      // if both values are the same
      if (xs[i] === xs[j]) {
        // increment the count
        cnt++;
      }
    }

    // if the current count is not divisible by 2
    if (cnt % 2 !== 0) {
      // set the return value to the current value
      retVal = xs[i];
      break;
    }
  }

  // return the value with an odd number of occurences
  return retVal;
};

File: c++/6kyu/FindTheOddInt.cpp
/*
Given an array of integers, find the one that appears an odd number of times.

There will always be only one integer that appears an odd number of times.
*/

#include <vector>
#include <iostream>

int findOdd(const std::vector<int>& numbers){
  const int sz = numbers.size();
  int cnt = 0;
  int foundVal = 0;
  
  for (int i = 0; i < sz; i++) {
    cnt = 0;
    int curVal = numbers[i];
    for (int j = 0; j < sz; j++) {
      if (curVal == numbers[j]) {
        cnt++;
      }
    }
    
    if (cnt % 2 != 0) {
      foundVal = curVal;
      break;
    } else {
      continue;
    }
  }
  return foundVal;
}
